src/main/java/com/mongodb/hibernate/jdbc/MongoConnection.java

```diff
@@ -0,0 +1,221 @@
+    }
+
+    @Override
+    public PreparedStatement prepareStatement(String mql, int resultSetType, int resultSetConcurrency) {
+        throw new NotYetImplementedException(
+                "To be implemented in scope of https://jira.mongodb.org/browse/HIBERNATE-13");
+    }
+
+    @Override
+    public PreparedStatement prepareStatement(String mql, int autoGeneratedKeys) {
+        throw new UnsupportedOperationException("Auto-generated key from MongoDB server side unsupported");
+    }
+
+    @Override
+    public PreparedStatement prepareStatement(String mql, String[] columnNames) {
+        throw new UnsupportedOperationException("Auto-generated key from MongoDB server side unsupported");
+    }
+
+    // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    // SQL99 data types
+
+    @Override
+    public Clob createClob() {
+        throw new NotYetImplementedException();
+    }
+
+    @Override
+    public Blob createBlob() {
+        throw new NotYetImplementedException();
+    }
+
+    @Override
+    public NClob createNClob() {
+        throw new NotYetImplementedException();
+    }
+
+    @Override
+    public SQLXML createSQLXML() {
+        throw new NotYetImplementedException();
+    }
+
+    @Override
+    public Array createArrayOf(String typeName, Object[] elements) {
+        throw new NotYetImplementedException();
+    }
+
+    @Override
+    public Struct createStruct(String typeName, Object[] attributes) {
+        throw new NotYetImplementedException();
+    }
+
+    // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    // DataBase meta data
+
+    @Override
+    public DatabaseMetaData getMetaData() {
+        throw new NotYetImplementedException(
+                "To be implemented in scope of https://jira.mongodb.org/browse/HIBERNATE-37");
+    }
+
+    // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    // unsupported operations
+
+    @Override
+    public CallableStatement prepareCall(String mql) {
+        throw new UnsupportedOperationException(
+                "MongoDB's JSON server function has been deprecated in favour of 'aggregate' pipeline");
+    }
+
+    @Override
+    public CallableStatement prepareCall(String mql, int resultSetType, int resultSetConcurrency) {
+        throw new UnsupportedOperationException(
+                "MongoDB's JSON server function has been deprecated in favour of 'aggregate' pipeline");
+    }
+
+    // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    // dummy implementations
+
+    @Override
+    public @Nullable SQLWarning getWarnings() {
+        return null;
+    }
+
+    @Override
+    public void clearWarnings() {
+        // no-op
+    }
+
+    @Override
+    public <T> T unwrap(Class<T> unwrapType) {
+        throw new UnknownUnwrapTypeException(unwrapType);
+    }
+
+    @Override
+    public boolean isWrapperFor(Class<?> unwrapType) {
+        return false;
+    }
```

Comment:
The above two won't be used as `MongoConnectionProvider`; the two `SQLWarning` related methods is used soley in Hibernate's `org.hibernate.engine.jdbc.spi.SqlExceptionHelper` class's `void handleAndClearWarnings()` method as below:
```
public void handleAndClearWarnings(
			Connection connection,
			WarningHandler handler) {
		try {
			if ( logWarnings ) {
				walkWarnings( **connection.getWarnings()**, handler );
			}
		}
		catch (SQLException sqle) {
			// workaround for WebLogic
			LOG.debug( "could not log warnings", sqle );
		}
		try {
			// Sybase fail if we don't do that, sigh...
			**connection.clearWarnings()**;
		}
		catch (SQLException sqle) {
			LOG.debug( "could not clear warnings", sqle );
		}
	}
```
and `org.hibernate.engine.jdbc.connections.internal.DriverManagerConnectionProviderImpl`'s following method:

```
protected Connection releaseConnection(Connection conn) {
			Exception t = null;
			try {
				conn.setAutoCommit( true );
				**conn.clearWarnings()**;
				if ( connectionValidator.isValid( conn ) ) {
					return conn;
				}
```

Seems we don't need to anything.